service/history/service.go

```diff
@@ -309,6 +314,7 @@ func NewConfig(dc *dynamicconfig.Collection, numberOfShards int, storeType strin
 		ReplicatorProcessorUpdateAckIntervalJitterCoefficient:  dc.GetFloat64Property(dynamicconfig.ReplicatorProcessorUpdateAckIntervalJitterCoefficient, 0.15),
 		ReplicatorProcessorRedispatchInterval:                  dc.GetDurationProperty(dynamicconfig.ReplicatorProcessorRedispatchInterval, 5*time.Second),
 		ReplicatorProcessorRedispatchIntervalJitterCoefficient: dc.GetFloat64Property(dynamicconfig.ReplicatorProcessorRedispatchIntervalJitterCoefficient, 0.15),
+		ReplicatorProcessorMaxRedispatchQueueSize:              dc.GetIntProperty(dynamicconfig.ReplicatorProcessorMaxRedispatchQueueSize, 10000),
```

Comment:
How much memory does a single task take ? Are all tasks equal ? If not, consider using a size based limit instead of a count based limit. Same comment for other queue processors.

Reply:
A single task takes less than 1KB (actually much smaller, but we can use this as an upper bound). All tasks will have similar size as they only contains some metadata information like domainID, workflowID, timestamp, task type, task state etc. So I think count based limit is fine here.

The number 10k here is a guess based on our current number of pending tasks number in staging/prod. I will need to tune this number when doing testing and after deployment.